katharsis-core/src/main/java/io/katharsis/utils/parser/TypeParser.java

```diff
@@ -105,7 +106,11 @@ public TypeParser() {
 		} else {
 			Method method;
 			try {
-				method = clazz.getMethod("parse", String.class);
+				try {
```

Comment:
Adding the reflection penalty on every single String parsed by this parser is highly unnecessary and undesirable.  If this code is intended to be invoked rather than copied and extended, then I second the use of the caching method @chb0github suggested.  A warm-up penalty is reasonable, but this should maintain higher throughput.

I am also concerned at the StringParser interface lacking the ability to set the encoding.